packages/components/src/Grid.astro

```diff
@@ -0,0 +1,219 @@
+---
+const { container, fluid, xs, sm, md, lg, auto, gap } = Astro.props;
+const containerClass = container ? "grid-container" : "";
+const autoClass = auto ? "auto-columns" : "";
+const gapClass = gap ? `gap-${gap}` : "";
+const xsClass = xs ? `col-${xs}` : "";
+const smClass = sm ? `col-sm-${sm}` : "";
+const mdClass = md ? `col-md-${md}` : "";
+const lgClass = lg ? `col-lg-${lg}` : "";
+const fluidClass = fluid ? "fluid" : "";
+const classList = [
+  containerClass,
+  fluidClass,
+  autoClass,
+  gapClass,
+  xsClass,
+  smClass,
+  mdClass,
+  lgClass,
+]
+  .join(" ")
+  .trim();
+---
+
+<style lang="scss">
+  .grid-container {
+    display: grid;
+    grid-template-columns: repeat(12, 1fr);
+    gap: 16px;
+    max-width: 1440px;
```

Comment:
Do we need a max width on this? It can also be set by the consumer if needed and we don't know what an appropriate value would be in the consuming application.